packager/signer/signer.go

```diff
@@ -464,15 +464,22 @@ func formatLinkHeader(preloads []*rpb.Metadata_Preload) (string, error) {
 	return strings.Join(values, ","), nil
 }
 
-// serveSignedExchange does the actual work of transforming, packaging and signed and writing to the response.
-func (this *Signer) serveSignedExchange(resp http.ResponseWriter, fetchResp *http.Response, signURL *url.URL, act string, transformVersion int64) {
+// serveSignedExchange_ConsumeCapCompliantFetchResp method consumes the gateway
+// response body, and checks if it's shorter than maxBodyLength and is therefore
+// OK to sign. It does so keeping memory consumption below maxBodyLength. It
+// returns ok iff succeeded and handles the errors if any. Specifically:
+//   * If failed to read the body, it propagates the error.
+//   * If hit the cap limit, it proxies the content unsigned.
+// In both problematic cases it returns false notifying serveSignedExchange to
+// cancel signing.
+func serveSignedExchange_ConsumeCapCompliantFetchResp(fetchBody *[]byte, resp http.ResponseWriter, fetchResp *http.Response) (success bool) {
```

Comment:
I'm not wild about this function name, but I don't have a better alternative. :shrug: Probably it's because the functionality is a bit hard to describe (three different cases). But I don't have a better alternative how to divide responsibilities between functions, either. :shrug:

Reply:
I felt the same. I think I now have a better idea, please have a look.